app/data_model/questionnaire_store.py

```diff
@@ -36,7 +35,7 @@ def __init__(self, user_id, user_ik):
             if 'COMPLETED_BLOCKS' in data:
                 self.completed_blocks = data['COMPLETED_BLOCKS']
 
-        self.initial_hash = hash(self.get_json())
+        self.data_on_load = self.get_json()
```

Comment:
Change Line 27 to return the raw JSON string then do the json.loads() in here, call it once for the 'live' data to edit and once for the data_on_load; then on has_changed check the two Python dicts